frontend/src/components/CartProductActions/CartProductActions.test.tsx

```diff
@@ -0,0 +1,211 @@
+        updateCartItemQuantity: updateCartItemQuantityMock,
+      } as unknown as CartContext.CartContextType);
+
+      renderWithRouter(
+        <CartProductActions cartProductItem={singleMockProduct} />,
+      );
+
+      const plusButton = screen.getByTestId("cart-product-increase");
+      await fireEvent.click(plusButton);
+
+      expect(updateCartItemQuantityMock).toHaveBeenCalledWith(
+        singleMockProduct.product.supermarket,
+        singleMockProduct.product.id,
+        singleMockProduct.quantity + 1,
+      );
+    });
+
+    it("Then using the trash button should remove the product from the cart", async () => {
+      const removeCartItemMock = vi.fn();
+
+      vi.spyOn(CartContext, "useCart").mockReturnValue({
+        removeCartItem: removeCartItemMock,
+      } as unknown as CartContext.CartContextType);
+
+      renderWithRouter(
+        <CartProductActions cartProductItem={singleMockProduct} />,
+      );
+
+      const trashButton = screen.getByTestId("cart-product-trash");
+      fireEvent.click(trashButton);
+
+      const confirmButton = await screen.findByRole("button", {
+        name: /confirm/i,
+      });
+      expect(confirmButton).toBeInTheDocument();
+
+      await fireEvent.click(confirmButton);
+
+      expect(removeCartItemMock).toHaveBeenCalledWith(
+        singleMockProduct.product.supermarket,
+        singleMockProduct.product.id,
+      );
+    });
+
+    it("Then using the minus button when the quantity is 1 should ask the user for deletion confirmation ", async () => {
+      const updateCartItemQuantityMock = vi.fn();
+
+      vi.spyOn(CartContext, "useCart").mockReturnValue({
+        updateCartItemQuantity: updateCartItemQuantityMock,
+      } as unknown as CartContext.CartContextType);
+
+      renderWithRouter(
+        <CartProductActions
+          cartProductItem={{
+            quantity: 1,
+            product: nwProduct,
+          }}
+        />,
+      );
+
+      const minusButton = screen.getByTestId("cart-product-minus");
+      await fireEvent.click(minusButton);
+
+      const trashConfirmButton = await screen.findByRole("button", {
+        name: /confirm/i,
+      });
+      expect(trashConfirmButton).toBeInTheDocument();
+    });
+  });
+
+  describe("When the user incorrectly inputs invalid data", () => {
+    let updateCartItemQuantityMock: ReturnType<typeof vi.fn>;
+    let quantityInput: HTMLInputElement;
+
+    beforeEach(() => {
+      updateCartItemQuantityMock = vi.fn();
+
+      vi.spyOn(CartContext, "useCart").mockReturnValue({
+        updateCartItemQuantity: updateCartItemQuantityMock,
+      } as unknown as CartContext.CartContextType);
+
+      renderWithRouter(
+        <CartProductActions cartProductItem={singleMockProduct} />,
+      );
+
+      quantityInput = screen.getByTestId(
+        "cart-product-input",
+      ) as HTMLInputElement;
+    });
+
+    it("Then using the manual input with an value of 0 should update the cart total to 1", async () => {
```

Comment:
```suggestion
    it("Then using the manual input with an value of 0 updates the cart total to 1", async () => {
```
And same comment as previously about `manual input`